sample/src/main/java/org/walletconnect/example/wallet/scanner/ScannerFragment.kt

```diff
@@ -0,0 +1,143 @@
+        setupObserver()
+    }
+
+    private val cameraPermissionCallback =
+        registerForActivityResult(ActivityResultContracts.RequestPermission()) { isGranted ->
+            if (isGranted) {
+                bindCameraUseCases()
+            }
+        }
+
+    private fun setupObserver() {
+        viewModel
+            .processCameraProvider
+            .observe(viewLifecycleOwner, { provider: ProcessCameraProvider ->
+                cameraProvider = provider
+                if (isCameraPermissionGranted()) {
+                    bindCameraUseCases()
+                } else {
+                    cameraPermissionCallback.launch(Manifest.permission.CAMERA)
+                }
+            })
+    }
+
+    private fun bindCameraUseCases() {
+        bindPreviewUseCase()
+        bindAnalyseUseCase()
+    }
+
+    private fun bindPreviewUseCase() {
+        cameraProvider?.let {
+            if (previewUseCase != null) {
+                it.unbind(previewUseCase)
+            }
+
+            previewUseCase = Preview.Builder()
+                .setTargetRotation(binding.previewView.display.rotation)
+                .build()
+            previewUseCase?.setSurfaceProvider(binding.previewView.surfaceProvider)
+            cameraProvider?.bindToLifecycle(this, cameraSelector!!, previewUseCase)
+        }
+    }
+
+    private fun bindAnalyseUseCase() {
+        BarcodeScannerOptions.Builder()
+            .setBarcodeFormats(Barcode.FORMAT_QR_CODE)
+            .build();
+        val barcodeScanner: BarcodeScanner = BarcodeScanning.getClient()
+
+        cameraProvider?.let {
+            if (analysisUseCase != null) {
+                it.unbind(analysisUseCase)
+            }
+
+            analysisUseCase = ImageAnalysis.Builder()
+                .setTargetRotation(binding.previewView.display.rotation)
+                .build()
+
+            analysisUseCase?.setAnalyzer(Executors.newSingleThreadExecutor(),
+                { imageProxy -> processImageProxy(barcodeScanner, imageProxy) }
+            )
+
+            cameraProvider?.bindToLifecycle(this, cameraSelector!!, analysisUseCase)
+        }
+    }
+
+    @SuppressLint("UnsafeOptInUsageError")
+    private fun processImageProxy(barcodeScanner: BarcodeScanner, imageProxy: ImageProxy) {
+        val inputImage =
+            imageProxy.image?.let {
+                InputImage.fromMediaImage(
+                    it,
+                    imageProxy.imageInfo.rotationDegrees
+                )
+            }
+        inputImage?.let { image ->
+            barcodeScanner.process(image)
+                .addOnSuccessListener { barcodes ->
+                    if (barcodes.isNotEmpty() && shouldScan) {
+                        barcodes.first().rawValue?.let {
+                            shouldScan = false
+                            viewModel.pair(it)
+                            activity?.onBackPressed()
```

Comment:
Since the navigation component is being used, replace this with popBackStack()
```suggestion
findNavController().popBackStack()
```